src/utils/vectorEncoder.ts

```diff
@@ -385,11 +343,10 @@ function differenceDigit(num: string) {
 
 /**
  * Convert number to the DUP format, using pseudo digits.
- * @return {string}
  */
 function duplicateDigit(num: string) {
   let DUPdigit = '';
-  DUPdigit += pseudoDigits[DUP][Number(num.charAt(0))];
+  DUPdigit += pseudoDigits[DUP][num.charCodeAt(0) - 48];
```

Comment:
Please use camelcase. dupDigit, diffDigit, ...

And seems to me there is more than one digit (even if technically it can be letter + some digits) so it should be plural.